test/middleware/AuthMiddleware.test.ts

```diff
@@ -0,0 +1,127 @@
+import 'reflect-metadata';
+import '../../src/controllers/index';
+import { expect, assert } from 'chai';
+import * as request from 'supertest';
+import { PENALTY_DETAILS_PAGE_URI, OTHER_REASON_DISCLAIMER_PAGE_URI, OTHER_REASON_PAGE_URI } from '../../src/utils/Paths';
+import { createApp, getDefaultConfig } from '../ApplicationFactory';
+import { Maybe, VerifiedSession } from 'ch-node-session-handler';
+import { createFakeSession } from '../utils/session/FakeSessionFactory';
+import { SessionKey } from 'ch-node-session-handler/lib/session/keys/SessionKey';
+import { generateSessionId, generateSignature } from 'ch-node-session-handler/lib/utils/CookieUtils';
+
+const config = getDefaultConfig();
+const id = generateSessionId();
+const sig = generateSignature(id, config.cookieSecret);
+
+const session = createFakeSession(
+    [{
+        [SessionKey.Id]: id
+    }, {
+        [SessionKey.ClientSig]: sig
+    }], config.cookieSecret, true);
+
+const authedApp = createApp(session);
+
+const protectedPages = [
+    PENALTY_DETAILS_PAGE_URI,
+    OTHER_REASON_DISCLAIMER_PAGE_URI,
+    OTHER_REASON_PAGE_URI
+];
+
+describe('Authentication Middleware', () => {
+
+
+    describe('Authed path', () => {
+
+        let penaltyDetails: Record<string, any>;
+
+        before('sign in server should be running', async () => {
+
+            penaltyDetails = {
+                penaltyReference: 'A12345678',
+                companyNumber: 'SC123123'
+            };
+
+            await request(createApp())
+                .get(PENALTY_DETAILS_PAGE_URI)
+                .catch(err => assert.fail(err));
+
+
+        });
+
+        it('should not redirect the user to the sign in page if the user is signed in', async () => {
+
+            for (const page of protectedPages) {
+                await request(authedApp).get(page)
+                    .expect(200);
+            }
+
+        });
+
+        it('should allow the user to go to the penalty reference screen if authed', async () => {
+
+            await request(authedApp)
+                .get(PENALTY_DETAILS_PAGE_URI)
+                .expect(200);
+
+        });
+        it('should continue to the next page without requiring auth if penalty details are valid', async () => {
```

Comment:
There are few things I noticed about this test:
1) it targets penalty details page (integration) and it's behaviour while it should more focus on auth logic
2) it says 'if penalty details are valid' while validity of data does not matter at all since middleware is called before even controller gets the request and for that reason it is pointless

To fix it I suggest you change it to iterate through all the routes making POST request and assess that there is no redirect to sign in page.

Even better solution would be making this test a unit tests not integration tests but we have little time to change it.